src/Session.hs

```diff
@@ -125,6 +134,40 @@ sessionRestart session@Session{..} = do
     sessionStart session cmd setup
 
 
+performEvals :: Ghci -> Bool -> [FilePath] -> IO [Load]
+performEvals _ False _ = pure []
+performEvals ghci True reloaded = do
+  cmds <- traverse getCommands reloaded
+  fmap join $ flip traverse cmds $ \(file, cmds') ->
+    flip traverse cmds' $ \(num, cmd) -> do
+      ref <- newIORef []
+      execStream ghci (intercalate " " $ lines cmd) $ \_ resp -> modifyIORef ref (++ resp)
+      resp <- readIORef ref
+      pure $ EvalResult file (num, 1) cmd resp
+
+
+getCommands :: FilePath -> IO (FilePath, [(Int, String)])
+getCommands fp = do
+  ls <- readFile fp
+  pure (fp, splitCommands $ zip [1..] $ lines ls)
+
+splitCommands :: [(Int, String)] -> [(Int, String)]
+splitCommands [] = []
+splitCommands ((num, line) : ls)
+  | isCommand line =
```

Comment:
See the stripPrefix function which can avoid the drop and the isPrefixOf pair.

Reply:
Cool, I didn't know about this! I'm not entirely sure how to make the change though, since we still want to `span` over the next lines.